src/ducks/restaurantDetail.js

```diff
@@ -0,0 +1,51 @@
+import { call, put } from 'redux-saga/effects';
+import {
+  fetchRestaurantDetail,
+} from '../util/api';
+
+export const RESTAURANT_DETAIL_FETCH_REQUESTED = 'RESTAURANT_DETAIL_FETCH_REQUESTED';
+export const RESTAURANT_DETAIL_FETCH_SUCCEEDED = 'RESTAURANT_DETAIL_FETCH_SUCCEEDED';
+export const RESTAURANT_DETAIL_FETCH_FAILED = 'RESTAURANT_DETAIL_FETCH_FAILED';
+
+export function loadRestaurantDetail(id) {
+  return {
+    type: RESTAURANT_DETAIL_FETCH_REQUESTED,
+    payload: id,
+  };
+}
+
+export function loadRestaurantDetailSuccess(restaurantDetail) {
+  return {
+    type: RESTAURANT_DETAIL_FETCH_SUCCEEDED,
+    payload: restaurantDetail,
+  };
+}
+
+export function loadRestaurantDetailFailure(error){
+  console.error(error);
+  return {
+    type: RESTAURANT_DETAIL_FETCH_FAILED,
+    error: error.message,
+  };
+}
+
+export function* loadRestaurantDetailWorker(action) {
+  try {
+    const restaurantDetail = yield call(fetchRestaurantDetail, action.payload);
```

Comment:
Separate the api method to have to seaparate api requests. Then let the worker be responsible for merging the two results together and then dispatching that via `loadRestaurantDetailSuccess`